recipients-resolver/src/main/java/com/redhat/cloud/notifications/recipients/resolver/FetchUsersFromExternalServices.java

```diff
@@ -108,27 +107,48 @@ public List<User> getUsers(String orgId, boolean adminsOnly) {
 
         List<User> users;
         String supplier;
-        if (recipientsResolverConfig.isFetchUsersWithRbacEnabled()) {
-            Log.debug("Fetching users with RBAC");
-            supplier = "RBAC";
-            users = getWithPagination(
-                page -> retryOnError(() -> {
-                    LocalDateTime startTime = LocalDateTime.now();
-                    Page<RbacUser> rbacUserPage = rbacServiceToService.getUsers(orgId, adminsOnly, page * recipientsResolverConfig.getMaxResultsPerPage(), recipientsResolverConfig.getMaxResultsPerPage());
-                    Duration duration = Duration.between(startTime, LocalDateTime.now());
-                    if (recipientsResolverConfig.getLogTooLongRequestLimit().compareTo(duration) < 0) {
-                        Log.warnf("Rbac service response time was %ds for request OrgId: %s, adminOnly: %s, page %d ", duration.toSeconds(), orgId, adminsOnly, page);
-                    }
-                    return rbacUserPage;
-                }));
-        } else if (recipientsResolverConfig.isFetchUsersWithMbopEnabled()) {
-            Log.debug("Fetching users with BOP/MBOP");
-            supplier = "BOP";
-            users = fetchUsersWithMbop(orgId, adminsOnly);
-        } else {
-            Log.debug("Fetching users with IT service");
-            supplier = "IT";
-            users = fetchUsersWithItUserService(orgId, adminsOnly);
+
+        try {
+            if (recipientsResolverConfig.isFetchUsersWithRbacEnabled()) {
+                Log.debug("Fetching users with RBAC");
+                supplier = "RBAC";
+                users = getWithPagination(
+                    page -> retryOnError(() -> {
+                        LocalDateTime startTime = LocalDateTime.now();
+                        Page<RbacUser> rbacUserPage = rbacServiceToService.getUsers(orgId, adminsOnly, page * recipientsResolverConfig.getMaxResultsPerPage(), recipientsResolverConfig.getMaxResultsPerPage());
+                        Duration duration = Duration.between(startTime, LocalDateTime.now());
+                        if (recipientsResolverConfig.getLogTooLongRequestLimit().compareTo(duration) < 0) {
+                            Log.warnf("Rbac service response time was %ds for request OrgId: %s, adminOnly: %s, page %d ", duration.toSeconds(), orgId, adminsOnly, page);
+                        }
+
+                        return rbacUserPage;
+                    }));
+            } else if (recipientsResolverConfig.isFetchUsersWithMbopEnabled()) {
+                Log.debug("Fetching users with BOP/MBOP");
+                supplier = "BOP";
+                users = fetchUsersWithMbop(orgId, adminsOnly);
+            } else {
+                Log.debug("Fetching users with IT service");
+                supplier = "IT";
+                users = fetchUsersWithItUserService(orgId, adminsOnly);
+            }
+        } catch (final FailsafeException | WebApplicationException e) {
```

Comment:
The only thing I changed about this bit is surrounding it with a `try / catch` block.